docs/hugo/content/design/ADR-2023-02-Adoption-Policy.md

```diff
@@ -0,0 +1,258 @@
+- Uses a concept we already have.
+
+**Disadvantages:**
+- Possibility of collision between user specified `reconcile-policy` and the `reconcile-policy` written by the operator because
+  of `reconcile-policy-if-exists`. 
+  - We can mitigate this issue by rejecting some configurations with a webhook, although we have to be careful how we craft
+    that webhook because the operator itself needs to be able to set both. We can either just block both on create but allow
+    both on update, or we can block both on create/update but only as long as the finalizer hasn't been set. Once the finalizer 
+    has been set, changing `reconcile-policy-if-exists` has no effect (we've already applied whatever we needed to) and all changes
+    to the policy should flow via the `reconcile-policy` annotation.
+- Excess events generated by us setting `reconcile-policy`, which is an annotation we watch for updates. We can deal with this: 
+  - If the `reconcile-policy-if-exists` is `skip` this is safe as we never re-reconcile when `reconcile-policy` is being set to skip.
+  - If the `reconcile-policy-if-exists` is `detach-on-delete` we're safe as those events only matter for delete so also don't trigger events.
+
+
+#### An implementation note on fault tolerance
+
+If we choose either flavor of [option 4](#option-4--allow-configuration-at-the-operator-or-per-resource-level) we need to
+ensure that we are fault-tolerant. We must avoid a situation where the operator:
+
+- Checks if the resource in Azure exists and determines it does not
+- Creates the resource in Azure
+- Crashes or is restarted and does not ever persist to etcd the fact the resource didn't already exist
+- Restarts and thinks that the resource already existed, triggering the adoption logic on a resource ASO itself just created
+
+This means there must be an etcd write after checking if the resource exists, before we actually issue a PUT to Azure.
+The existing `Claim` step in the operator reconcile path already caters to exactly this problem. 
+
+Unfortunately because ARM resources don't universally support `If-Match` headers + `etag` we are open to a race where
+the resource is created between the GET and the following PUT. There's nothing we can do about that without optimistic 
+concurrency support on the server side.
+
+## Decision
+
+I propose we choose [option 1: Default `reconcile-policy` is `manage`](#option-1--the-default-reconcile-policy-for-adopted-resources-is-manage).
+As mentioned in the discussion of that topic, it's the default we currently have and so is non-breaking to adopt, and is how
+ARM templates/BICEP, raw REST and the SDKs work today already. Extra protection against adoption/updating a resource that
+already exists can be achieved with some of those technologies, but it's not the default and requires the user to opt-in
+to it.
+
+Alongside that, I propose we choose [option 4.2: `reconcile-policy-if-exists` annotation](#option-42--reconcile-policy-if-exists-annotation)
```

Comment:
This would be great to have for CAPZ and is basically exactly what I had in mind. Agreed that this approach seems better than `adoption-policy`.